backend/src/api/app.py

```diff
@@ -132,6 +133,18 @@ async def report(file: UploadFile):
         logger.exception(e)
         return JSONResponse(status_code=500, content=file_upload_failed_response)
 
+@app.get("/report/{id}")
+def download_report(id: str):
+    logger.info(f"Get report download called for id: {id}")
+    try:
+        final_result = get_report(id)
+        if final_result is None:
+            return JSONResponse(status_code=404, content=f"Message with id {id} not found")
+        headers = {'Content-Disposition': 'attachment; filename="report.md"'}
```

Comment:
We probably want to generate this filename, once we have the company name stored with the report.